### ISSUE TITLE
Address FOUT issue

### ISSUE THREAD

**Issue by axedre** [KyleAMathews/typefaces#93]
![fout](https://user-images.githubusercontent.com/3670063/46859792-0cd64380-ce0f-11e8-8a26-fbac4527c587.gif)

I am experiencing the so-called FOUT issue, described in [this article](https://webdesign.tutsplus.com/articles/quick-tip-avoid-fout-by-adding-a-web-font-preloader--webdesign-8287); tried the workaround of adding the `wf-loading` class to the `<html>` tag but to no avail, so I guess it's still not supported in your package. Any plan to implement this? Not super urgent I guess, but the flickering is quite annoying (even more so on large portions of text, where the overall width and height of the paragraph changes significantly when the font is rendered).

Cheers and thanks for the great work on porting the fonts to npm!

**Comment by axedre** [comment]
Upon more careful reading (I had only skimmed through, to be fair), it seems it would boil down to bundling [WebfontLoader](https://github.com/typekit/webfontloader) (or a subset of its functionality)

**Comment by dmitrizzle** [comment]
The issue you're mentioned is already addressable by WebfontLoader as a separate package **and** via new browsers' ability to understand `font-display: swap` rule which is already part of this package.

**Comment by axedre** [comment]
Could you elaborate on the latter? Not to keen on adding another package if `typeface-*` offers a valid solution.

**Comment by dmitrizzle** [comment]
Here's what the property does: https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display (spoiler: more or less everything that WebfontLoader does).
Here's which browsers support it: https://caniuse.com/#search=font-display

**Comment by KyleAMathews** [comment]
Yeah all gatsby does is set font-display to swap. If you don't want this (not recommended as it slows loading your site) then you can override the CSS.

See the CSS template file here https://github.com/KyleAMathews/typefaces/blob/master/scripts/templates.js

**Comment by axedre** [comment]
Ok thanks for the suggestion, I will read up more on the `font-display` property I didn't know about. Cheers.

**Comment by dougmacknz** [comment]
> Yeah all gatsby does is set font-display to swap. If you don't want this (not recommended as it slows loading your site) then you can override the CSS.

How exactly does one override the `font-display` property that the typeface libraries are setting? I can see the setting in the template you linked, but I don't understand how you go from doing `require("typeface-open-sans")` to adjusting that?

**Comment by imshuffling** [comment]
> https://github.com/KyleAMathews/typefaces/blob/master/scripts/templates.js

I'm not clear on how to override the css. The font files are nested in the public folder, with additional characters in the file names (presume this changes on each `gatsby build`?)

<img width="458" alt="Screenshot 2019-05-16 at 10 03 32" src="https://user-images.githubusercontent.com/739061/57841120-e243d300-77c1-11e9-9a08-3cc81e35a54c.png">

**Comment by zubko** [comment]
Here is my solution to the "override" part, in case if it will help to save some time for next people.

So each of the typefaces are basically 1 css file and some fonts files (not included on github but included on npm). And that css file has a bunch of @font-face declarations, and @font-face doesn't participate in cascading (according to https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade#Which_CSS_entities_participate_in_the_cascade), so the whole @font-face should be redefined.

I had 2 fonts, let's say 'font-a' and 'font-b'. 

1. I created a new file `fonts.css` with the contents of both main.css files for each font. 

1. Then for each @font-face declaration I fixed the path to the font file. It was smth like `./files/font-a-bla-bla` and it became `../../node_modules/typeface-font-a/files/font-a-bla-bla` (depending on the location of fonts.css in the project).

1. Then I replaced the imports of typefaces in the global file (for Gatsby it's `gatsby-browser.js`) with the import of my file, smth like `import './src/css/fonts.css';` 

And voila! the Webpack was happy in development and production and I could decide for myself if `font-display: swap;` was the best option or not :)
